prusti-contracts/src/lib.rs

```diff
@@ -87,4 +87,29 @@ pub fn old<T>(arg: T) -> T {
     arg
 }
 
+#[allow(unused_variables)]
+pub fn implication(lhs: bool, rhs: bool) -> bool {
+    true
+}
+
+#[allow(unused_variables, unused_mut)]
+pub fn entailment<F1, F2>(mut f: F1, req: F2) -> bool {
```

Comment:
Even if we keep this underconstrained signature for taking in closure-like arguments, the `mut` doesn't do anything and can be removed (+ `unused_mut`).

Reply:
Doesn't this allow user closures which might mutate their environment to be specified with an entailment?

Reply:
No, that would be the difference between e.g. an `Fn(...)` type and an `FnMut(...)` type. Here we are not specifying either of those types, instead using a more generic `F1` type parameter. The `mut` modifier on the argument simply means that `f` can be reassigned inside the body of this `entailment` function, but the body of the function is irrelevant—we will match on the `entailment` call later on in the encoder instead.

Reply:
Oops. Fixed in 681ccfe0b2aa35aa9ed35a05b55490ba69ad34b9